File: python/injection/safe_query.py
import sqlite3

from flask import request

from injection.validators import sanitize_sort_column


def list_users_sorted(conn: sqlite3.Connection):
    raw_sort = request.args.get("sort", "id")
    sort_column = sanitize_sort_column(raw_sort)
    query = f"SELECT id, name FROM users WHERE active = ? ORDER BY {sort_column}"
    cur = conn.cursor()
    cur.execute(query, (True,))
    return cur.fetchall()

File: csharp/injection/SafeQuery.cs
using System.Data.SqlClient;

namespace DiffTest.Injection
{
    public class SafeUserRepository
    {
        public SqlCommand ListUsersSorted(string rawSort)
        {
            var sortColumn = SortColumnValidator.Sanitize(rawSort);
            return new SqlCommand($"SELECT Id, Name FROM Users ORDER BY {sortColumn}");
        }
    }
}
